src/contract/assembled_transaction.ts

```diff
@@ -754,11 +755,22 @@ export class AssembledTransaction<T> {
                 entry.credentials().address().signature().switch().name ===
                 "scvVoid"),
           )
-          .map((entry) =>
-            StrKey.encodeEd25519PublicKey(
-              entry.credentials().address().address().accountId().ed25519(),
-            ),
-          ),
+          .map((entry) => {
+            const addr = entry.credentials().address().address();
+            const addrType = addr.switch().name;
+            switch(addrType) {
+              case 'scAddressTypeAccount':
+                return StrKey.encodeEd25519PublicKey(
+                  addr.accountId().ed25519(),
+                );
+              case 'scAddressTypeContract':
+                return Address.contract(
+                  addr.contractId()
+                ).toString()
+              default:
+                throw new Error(`Unknown address type: ${addrType}`);
+            }
+          })
```

Comment:
Oh wait, you can just use [`Address.fromScAddress`](https://stellar.github.io/js-stellar-sdk/Address.html#fromScAddress) here!

Reply:
So much better! Thanks!